src/index.js

```
@@ -54,226 +56,301 @@ function buildNametoIdMap(objectArr, type) {
+      );
 
-    // upload artifact
-    if (uploadTestArtifact === "true") {
-      core.info("Uploading test api output...");
-      await artifactClient.uploadArtifact(
-        "transformer-test-results",
-        testOutputFiles,
-        "."
+      testOutputFiles.push(
+        `${testOutputDir}/${transformationHandleName}_diff.json`
       );
     }
+  }
+}
+
+// Upload the test results to an artifact store.
+async function uploadTestArtifacts(testOutputFiles) {
+  // upload artifact
+  if (uploadTestArtifact) {
+    core.info("Uploading test api output");
+    await artifactClient.uploadArtifact(
+      "transformer-test-results",
+      testOutputFiles,
+      "."
+    );
+  }
+}
+
+// Publish the transformations and libraries.
+async function publishTransformation(
+  transformationTest,
+  librariesTest,
+  commitId
+) {
+  // publish
+  res = await publish(transformationTest, librariesTest, commitId);
+  core.info(`Publish result: ${JSON.stringify(res.data)}`);
+}
+
+function colorize(message, color) {
+  const colors = {
+    reset: "\x1b[0m",
+    red: "\x1b[31m",
+    green: "\x1b[32m",
+    yellow: "\x1b[33m",
+    blue: "\x1b[34m",
+    magenta: "\x1b[35m",
+    cyan: "\x1b[36m",
+    white: "\x1b[37m",
+  };
+
+  return `${colors[color]}${message}${colors.reset}`;
+}
+
+// Log failed tests
+function logFailedTests(failedTests) {
+  core.info(colorize("\nFailed Tests:", "yellow"));
+  for (let i = 0; i < failedTests.length; i++) {
+    const test = failedTests[i];
+
+    core.info(colorize(`   ID: ${test.id}`, "red"));
+    core.info(colorize(`   Name: ${test.name}`, "red"));
+    core.info(colorize(`   Error: ${JSON.stringify(test.result)}\n`, "red"));
+    core.info("\n" + "=".repeat(40) + "\n"); // Add a line of equal signs between logs
+  }
+}
+
+async function testAndPublish() {
+  const transformationDict = {};
+  const libraryDict = {};
+  const outputMismatchResults = [];
+  const testOutputFiles = [];
+
+  try {
```

Let's fail fast here ?